test/unit/error.spec.ts

```diff
@@ -292,3 +285,97 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+      try {
+          throw { x: "Hello error object!" };
+      } catch (error) {
+          return error.x;
+      }
+    `.expectToMatchJsResult();
+});
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as classes",
+    errorType => {
+        util.testFunction`
+            try {
+                throw new ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw TypeError();
+                }
+            }
+        `
+            .expectNoExecutionError()
+            .expectToMatchJsResult();
+    }
+);
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as functions",
+    errorType => {
+        util.testFunction`
+            try {
+                throw ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
+                } else {
+                    throw TypeError();
+                }
+            }
+        `
+            .expectNoExecutionError()
+            .expectToMatchJsResult();
+    }
+);
+
+test("subclass Error", () => {
+    util.testFunction`
+        class MyError extends Error {
+            name: "MyError"
+        }
+        
+        try {
+            throw new MyError();
+        } catch (error) {
+            if (error instanceof Error) {
+                return \'\$error\';
+            } else {
```

Comment:
Only one branch of this if/else seems to do anything in this test?

Reply:
The else branch is implicitly stopping the test from giving false positives as I'm using `.expectNoExecutionError().expectToMatchJsResult()` to validate the result.

Reply:
Would be nice if you added a comment with that explanation in the code here, maybe in combination with a better error, something like `Error("This error serves to prevent false positives from .expectNoExecutionError()")`

Reply:
If it wouldn't throw an error it should fail anyway since result wouldn't be the same as in JS

Reply:
Assuming the test is properly written and works as expected. At this point It’s probably better to also check the actual result.

Reply:
We have discussed that with @Perryvw some time ago in Discord. The thing is that we don't really care about actual result of executed code, but only that it matches what JS would give. Ensuring that test code is correct was one of my worries about that, but adding expected result just gives one more point to make a mistake. Also, to check test code during development there is a `.debug` utility.

Reply:
Having an additional sanity check inside the test code is probably won't hurt then.